### DISCUSSION TITLE
Ability to group posts by year/month/day

### DISCUSSION THREAD

**Discussion by eryc-cc** [withastro/roadmap#387] [Proposal]
## Summary

The current behavior is to map all posts and display them one by one:

- Nov 11 - Post Name
- Nov 11 - Another Post
- Nov 10 - Yet Another Post

It would be nice to have the ability to group posts following a hierarchy:

- 2022
  - Nov
    - 11
      - Post Name
      - Another Post
    - 10
      - ...
  - Oct
    - ...
  - ...
- 2021
  - ...

**Comment by fflaten** [comment]
You can do this already. Either by using static routes where you place md/mdx files in folders under `/src/pages` by year, month etc which becomes part of the path. Or by using dynamic routing with `/posts/[year]/[month]/[day]/[slug].astro` or `/posts/[...path].astro`. I the latter you'd use `Astro.glob` to get all markdown-file and define the params from filename or frontmatter.

See https://docs.astro.build/en/core-concepts/routing/#example-dynamic-pages-at-multiple-levels

**Reply by eryc-cc** [reply]
Sorry, what I meant was to list all the posts in one page (like a blog home), but filter them by date, so all the posts from November 11 show up under a headline "November 11", and then all the posts from November 10 show up under "November 10".

Maybe what I need is to create a javascript that maps the posts, and if they're different dates, just add a headline with the date before the post?

**Reply by fflaten** [reply]
That's something you could solve with some JS/TS and markup in a page or a component, no framework-feature necessary.

**Comment by florian-lefebvre** [comment]
Yes there's nothing to be done in Astro for this, this is just a matter of implementation on your side
